sourcegraph/sourcegraph
# Code Insights: Add search context filter v1

Closes https://github.com/sourcegraph/sourcegraph/issues/33830
Closes https://github.com/sourcegraph/sourcegraph/issues/32963

## Background 

This PR updates our dilldown panel and adds a new control for the search context filter. It also fixes a few visual problems that we've been noticing recently about drilldown panel.

| Before | After |
| -- | -- |
| <img width="939" alt="Screenshot 2022-04-28 at 16 20 20" src="https://user-images.githubusercontent.com/18492575/165710802-c7166510-72c8-4c0b-98e8-299db9f8ccae.png"> | <img width="939" alt="Screenshot 2022-04-28 at 16 20 05" src="https://user-images.githubusercontent.com/18492575/165710918-d7fae7c5-8e30-44d6-9b31-919fdfd99b82.png"> |

## Todo 
- [x] Implement UI and the logic around context filter
- [x] Pass design review 
- [x] Pass tech review 
- [ ] Update our integration test in order to test search context filter [Seperate PR]

## Test plan

- Make sure that all integration tests work properly with 
- Make sure all visual details look correct according to the search UI [designs](https://www.figma.com/file/Ml5dtV9ni4mfffbfOfF9jg/Allow-search-contexts-as-filters-to-code-insights-(v0-%2333249-%26-v.1-%2333830)-%5Bready-for-dev%5D?node-id=1313%3A24762) 





## App preview:

- [Web](https://sg-web-vk-add-search-context-filter-v1.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-wranezedrg.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
